src/utils/env.ts

```diff
@@ -21,11 +20,23 @@ const PC_ENV_TLDS = environments
         return tlds;
     }, [] as string[])
     .concat(__PC_DEV_ENVS__);
+const GC_DEV_EXTRA_ENVS = environments.concat(__GC_DEV_EXTRA_ENVS__ as Environment[]);
 
 const [defaultEnv] = environments.filter(env => env.region === DEFAULT_ENV_REGION);
 
 export const DEFAULT_PC_ENV = buildPcEnv(defaultEnv.publicDomainName);
 
+function findPcEnvironment(hostname: string, targetEnv: string, envList: Environment[]): PcEnv|null {
```

Comment:
if it weren't for the extra domains, this would be super simple.  we could do:

```typescript
// parse is exported from sd-web
const parsedEnv = parse(hostname, {});
if (!parsedEnv || parsedEnv.name !== targetEnv) {
  return null;
  }
  return {
    pcEnvTld: parsedEnv.publicDomainName,
    pcAppOrigin: hostname
    }
```

This would achieve a few things:
1. built-in hardened parsing of hostnames to envs from sd-web
1. asserts that the provided hostname is in fact associated witht he targetEnv
1. would use the provided hostname which is preferred over us having to assume the subdomain is `apps`.

Reply:
the wrinkle though is of course the custom envs.  `parse` provides no option to customize the list of envs to search on.  So, we're never going to get the custom envs in parse.

I guess one option would be to use the default `parse` method (maybe down in `lookupGcEnv`) and then fall back to a more lenient parse that only considers the custom envs?  It is for dev anyway.

Thoughts?  We can also chat through this tomorrow.

Reply:
One other solution might be to add an extra `extra` boolean property for dev envs (added when using `.concat(`) .
That way we could use the parse method, or this custom one for extra envs.

Reply:
i do like the `GCEnvironment` and `extra` approach, but now, i'm realizing we don't even need to merge in envs and the __GC_DEV_EXTRA_ENVS__ into ALL_GC_ENVS.

`parse` will internally loop through it's known envs searching for matches.  So, we can call parse first and if it doesn't find an env, we can fall back to searching through the extra envs.  And, I think we can just change the `findPcEnvironment` method signature to just take a list of the extra envs, negating the need for the extra prop.  It's also a bit cleaner.

Final point, instead of trying to match `apps.${p}`, we should switch it to location.hostname.endsWith(p).  This will allow us to match any subdomain (i.e. apps or app).  Then, we can return

```
return {
pcEnvTld: environment.publicDomainName,
pcAppOrigin: location.origin
}
```

This gives us full support of localhost, extra envs, and match any subdomain (apps, app, etc) 🤘